src/App.tsx

```diff
@@ -15,37 +17,97 @@
   'Garlic',
 ];
 
+export const BY_ALPHABETICALLY = 'alphabetically';
+export const BY_LENGTH = 'length';
+export const BY_REVERSE = 'reverse';
+
+type SortField = typeof BY_ALPHABETICALLY | typeof BY_LENGTH | null;
+
+export function getSortList(
+  goods: string[],
+  sortField: SortField,
+  reverse: boolean,
+): string[] {
+  const sortGoods = [...goods];
+
+  switch (sortField) {
+    case BY_ALPHABETICALLY:
+      sortGoods.sort();
+      break;
+    case BY_LENGTH:
+      sortGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    sortGoods.reverse();
+  }
+
+  return sortGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField>(null);
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const sortGoods = (field: SortField | typeof BY_REVERSE) => {
+    if (field === BY_REVERSE) {
+      setReverse(prevReverse => !prevReverse);
+    } else {
+      setSortField(field);
+    }
+  };
+
+  const resetSorting = () => {
+    setSortField(null);
+    setReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== BY_ALPHABETICALLY,
+          })}
+          onClick={() => sortGoods(BY_ALPHABETICALLY)}
+        >
```

Comment:
To improve UX, consider adding an `aria-pressed` attribute to the buttons to indicate whether they are currently active. This would help users understand which sorting option is currently applied.